docs/PORTING.md

```diff
@@ -0,0 +1,40 @@
+
+The full list of Alpine packages required by EVE is available under [pkg/alpine](../pkg/alpine) and while it is not long, it actually requires twice as many packages available during the build phase so your bootstrapping efforts may take some time. The most convenient way to move from a cross-compiling `bootstrap.sh` phase of Alpine bringup to bringing up the rest of the packages is to use [qemu user emulation mode](https://qemu.readthedocs.io/en/latest/user/index.html) under either Docker desktop or [proot](https://proot-me.github.io/). Both of these will require an image of a filesystem that is complete enough to run basic Unix commands. A good place to star assembling this image into a single tarball is the output packages from the `bootstrap.sh`. Note, however, that you may need to manually untar them and put everything in just the right order yourself. Taking a look at what files comprise a typical Alpine docker image `docker pull alpine:latest` will give you a hint at what's required. Here's a minimal list of packages that need to appear in that filesystem image: `abuild curl tar make linux-headers patch g++ git gcc ncurses-dev autoconf`.
+
+Once you assembled your new image and launched a shell inside of that filesystem, you can start using standard Alpine's [abuild](https://wiki.alpinelinux.org/wiki/Abuild_and_Helpers) tools with the source code from [aports](https://github.com/alpinelinux/aports) to build the remainder of the required packages. The process involves cd'ing into `{main,community,testing}/package-name` folder in the aports tree, sometimes editing APKBUILD script and running `abuild -r`. Your resulting packages will get deposited under `$HOME/packages` provided that you're using a standard Alpine build environment with a dedicated builder user. A minimal set of steps required to setup such an environment is captured in the following [Dockerfile](../build-tools/src/scripts/Dockerfile.alpine.bootstrap). That same file also contains a skeleton of a dummy APKBUILD file that one can use to stub-out packages that are required by `pkg/alpine` but are not available yet.
+
+Once you have an MVP collection of Alpine packages you need to upload them plus the tarball of your build environment to some publicly available https endpoint (AWS S3 buckets work great).
```

Comment:
I am confused by this. What do we need it for? We can build packages and then cache them in an image.

Reply:
Because with the packages coming from the cross-build side of Alpine -- you can't -- there's nothing to get them from and building them as part of our infrastructure would be way too disruptive and confusing to EVE. So, again, these are part of notes on how to bootstrap just enough Alpine in parallel with using it in EVE right away.